src/hooks/use-speech-recognition.ts

```diff
@@ -0,0 +1,61 @@
+import { useRef, useState } from 'react';
+import { CapturedSpeech } from '../models/captured-speech';
+import { DeckLanguage, getLanguageCode } from '../models/language';
+
+export function useCaptureSpeech() {
+  const speechRecognitionRef = useRef(getSpeechRecognitionObj());
+  const [isCapturingSpeech, setIsCapturingSpeech] = useState(false);
+
+  return { captureSpeech, stopCapturingSpeech, abortSpeechCapture, isCapturingSpeech };
+
+  function captureSpeech(language: DeckLanguage) {
+    speechRecognitionRef.current.lang = getLanguageCode(language);
+    speechRecognitionRef.current.start();
+
+    return new Promise<CapturedSpeech>((resolve, reject) => {
+      setIsCapturingSpeech(true);
+
+      // Will only fire if the system can make a about the transcript above the confidence threshold
+      speechRecognitionRef.current.onresult = (event) => {
+        const resultsLength = event.results.length - 1;
+        const ArrayLength = event.results[resultsLength].length - 1;
+        const { transcript, confidence } = event.results[resultsLength][ArrayLength];
+        resolve({ transcript, confidence });
+      };
+
+      // Always fires when disconnected
+      // TODO: Find a guarantee that it will always fire after onresult
+      // Or find another to handle resolving the promise if the user has not spoken
```

Comment:
suggestion: we could pass in some type of timeout which is composed of `card audio length` + `the pause length` and resolve if it exceeds that.

Reply:
hmm nvm, i guess manually calling `stopCapturingSpeech` after the wait is basically doing the same thing

Reply:
Yeah, this one is tricky. It seems to be working now, but Ideally we would want some guarentees